Lib/fontTools/ttLib/ttGlyphSet.py

```diff
@@ -29,6 +35,17 @@ def __init__(self, font, location, glyphsMapping):
                 )
             # TODO VVAR, VORG
 
+    def pushLocation(self, location, reset: bool):
+        self.locationStack.append(self.location)
+        if reset:
+            self.location = self.originalLocation.copy()
+        else:
+            self.location = self.location.copy()
+        self.location.update(location)
+
+    def popLocation(self):
```

Comment:
Consider making pushLocation() a context manager with the `contextlib.contextmanager` decorator.

Reply:
Done. Thanks.